src/lib.rs

```diff
@@ -3848,6 +3848,50 @@ impl Connection {
         stream.is_readable()
     }
 
+    /// Returns true if the stream has enough send capacity.
+    ///
+    /// When `len` more bytes can be buffered into the given stream's send
+    /// buffer, `true` will be returned, `false` otherwise.
+    ///
+    /// In the latter case, if the additional data can't be buffered due to
+    /// flow control limits, the peer will also be notified.
+    ///
+    /// If the specified stream doesn't exist (including when it has already
+    /// been completed and closed), the [`InvalidStreamState`] error will be
+    /// returned.
+    ///
+    /// In addition, if the peer has signalled that it doesn't want to receive
+    /// any more data from this stream by sending the `STOP_SENDING` frame, the
+    /// [`StreamStopped`] error will be returned.
+    ///
+    /// [`InvalidStreamState`]: enum.Error.html#variant.InvalidStreamState
+    /// [`StreamStopped`]: enum.Error.html#variant.StreamStopped
+    #[inline]
+    pub fn stream_writable(
+        &mut self, stream_id: u64, len: usize,
+    ) -> Result<bool> {
```

Comment:
It might be nice to return which window is too small. Thinking purely for audit purpose. Rather than complicate error type, could substitute bool for an enum like {Ok, StreamBlocked, ConnectionBlocked}.

Reply:
That seems like a fairly niche use case, so I'd rather keep this API simple. But we could add `max_tx_data` / `max_rx_data` to `Stats` so that an application can figure this out on its own if it wants to (we already have cwnd there as well).

Reply:
Yeah that's cool